velox/docs/develop/types.rst

```diff
@@ -2,61 +2,75 @@
 Types
 =====
 
-Velox supports scalar types and complex types. These types cover most of the
-Presto and Spark data types.
+Velox supports a fixed set of physical types, an extensible set of logical types,
+complex types, and few types specific to Presto.
 
-Scalar Types
-~~~~~~~~~~~~
-Scalar types in Velox are logical and SQL-compatible.
-Each scalar type is implemented using a C++ type. The table
-below shows the supported scalar types and their corresponding C++ type.
+
+Physical Types
+~~~~~~~~~~~~~~
+Each physical type is implemented using a C++ type. The table
+below shows the supported physical types, their corresponding C++ type,
+and bytes required per value.
 
 ======================  ===========================    ==================
-Velox Type              C++ Type                       Bytes per Value
+Velox Physical Type     C++ Type                       Bytes per Value
 ======================  ===========================    ==================
 BOOLEAN                 bool                            0.125 (i.e. 1 bit)
 TINYINT                 int8_t                          1
 SMALLINT                int16_t                         2
 INTEGER                 int32_t	                        4
 BIGINT                  int64_t                         8
+HUGEINT                 int128_t                       16
 DATE                    struct Date                     8
 REAL                    float                           4
 DOUBLE                  double                          8
-DECIMAL                 int64_t / int128_t              8 / 16
 TIMESTAMP               struct Timestamp               16
-INTERVAL DAY TO SECOND  int64_t                        8
 VARCHAR                 struct StringView              16
```

Comment:
Should we clarify that VARCHAR and VARBINARY values are stored in 2 places: 16 bytes in the 'values' buffer + variable number of bytes in string buffers?